src/pykeen/stoppers/early_stopping.py

```diff
@@ -185,7 +185,7 @@ def best_metric(self) -> float:
         return self._stopper.best_metric
 
     @property
-    def best_epoch(self) -> int:
+    def best_epoch(self) -> Optional[int]:
```

Comment:
are there any implications for this later? does type checking not catch this?

Reply:
In the training loop, there is a check against `None` to see whether the stopper has been evaluated.